legacy/src/app/controllers/node_details.js

```diff
@@ -6,6 +6,38 @@
 
 import { HardwareType, NodeTypes } from "../enum";
 
+// Convert MiB to GiB value if over 1024 and round to 4 significant figures.
+const formatNumaMemory = (mib) => {
+  if (mib >= 1024) {
+    return `${Number((mib / 1024).toPrecision(4)).toString()} GiB`;
+  }
+  return `${Number(mib.toPrecision(4)).toString()} MiB`;
+};
+
+// Convert array of numbers into range strings,
+// e.g [0, 1, 2, 4, 6, 7] => ["0-2", "4", "6-7"]
+const getRanges = (array) => {
```

Comment:
Is the provided array always sorted low to high?

Reply:
I've assumed yes just because that's how the data's come in like that every time, but you're right I should sort it first lest we end up in an infinite loop.